src/KDSoapClient/KDSoapMessageAddressingProperties.cpp

```diff
@@ -419,7 +419,7 @@ void KDSoapMessageAddressingProperties::readMessageAddressingProperty(const KDSo
         KDSoapMessageRelationship::Relationship relationship;
         relationship.uri = (value.value().toString());
         relationship.relationshipType = addressingNS + QLatin1String("/reply");
-        foreach (KDSoapValue attr, value.childValues().attributes()) {
+        for (KDSoapValue attr : value.childValues().attributes()) {
```

Comment:
This one happens at runtime and should therefore be ported to a const local container.

Reply:
Actually it does return a `KDSoapValueList &` for `childValues()` and `const QList<KDSoapValue> &` for `attributes()`.
Does the deep copy still occure then?

Reply:
Ah OK, sorry, I didn't dig into this one. Indeed the ref solves the issue. Returning a const ref is not a great practice in itself, so I didn't think this was happening here.